src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py

```diff
@@ -271,14 +271,16 @@ def __call__(
         remainder = num_frames % self.stride
         if remainder != 0:
             input_features = input_features[:, :num_frames, :]
-            attention_mask = attention_mask[:, :num_frames]
+            if return_attention_mask:
+                attention_mask = attention_mask[:, :num_frames]
```

Comment:
How expensive is having `return_attention_mask=True` in the `pad` call? The logic here could all be tidied up by having it as `True` on L261 and then only keep it in `padded_inputs` on 264 if `return_attention_mask=True`